Test infrasturcture (issue #4) and direct console print of example (issue #5)

WoVollmer opened the pull request:
testfile: test_error_cases => running
testfile: test_use_cases => are running within testfile
-  but test file not running with devtools::test() or covr::package_coverage()
        throws error: object 'code' not found 
      reason not yet found => commented out the test cases

fabian-s committed:
"issue #4: Test infrastructure and first tests generated"
"issue #5: direct console print out of 'get_stability_paths() example'

Commit:
Merge branch 'master' of https://github.com/WoVollmer/stabpath

fabian-s commented:
Thx, but can't accept this yet. 

Please:

- clean up your commit history by doing "git rebase", see https://gist.github.com/Chaser324/ce0505fbed06b947d962#submitting-a-pull-request

- please fix the failing tests in `test_use_cases` -- the problem is that `model$call` when `refit.regsubsets` is run under `devtools::test` does not have the expected structure (you can check this yourself by inserting a `browser()` statement inside one of the failing `test_that` calls and inspecting the `model`object.) The way to fix it is to manually set `model$call <- quote(regsubsets(max_formula, data=data, nbest = 1, nvmax = 8,
  really.big = TRUE))` after l. 11.  

Please do so and commit your changes, the PR then updates automatically. Your  commit message should contain something like "fixes #4 " so that merging the PR will close the issue automatically.

Commit:
fixes "issue #4": Test infrastructure and first tests generated

fabian-s commented:
I stil get:
```
* checking for unstated dependencies in ‘tests’ ... WARNING
'library' or 'require' call not declared from: ‘testthat’
```
and, even worse :
```
* checking tests ...
  Running ‘testthat.R’
 ERROR
Running the tests in ‘tests/testthat.R’ failed.
Last 13 lines of output:
     TRUE, FALSE, TRUE, TRUE)), .Names = c("lcavol", "lweight", "age", "lbph", "svi",
     "lcp", "gleason", "pgg45", "lpsa", "train"), row.names = c("60", "41", "95", "35",
     "73", "38", "36", "29", "91", "77", "93", "2", "3", "47", "69", "74", "56", "53",
     "32", "61", "92", "76", "83", "37", "81", "42", "13", "9", "84", "65", "27", "70",
     "20", "14", "26", "5", "79", "78", "52", "86", "51", "40", "39", "48", "89", "23",
     "62", "68", "88"), class = "data.frame"), nbest = 1, nvmax = 8, really.big = TRUE)

  Error in (function (x, ...)  : object 'max_formula' not found
  Calls: test_check ... refit -> refit.regsubsets -> do.call -> <Anonymous>
  testthat results ================================================================
  OK: 3 SKIPPED: 0 FAILED: 1
  1. Error: get_stability_paths with prostate data - subsample (@test_use_cases.R#21)

  Error: testthat unit tests failed
  Execution halted
```

Please fix those, we need a package that passes `R CMD check --as-cran`  without NOTES, ERRORS, WARNINGS (at least; repair the ERRORS and WARNINGS related to #4)

fabian-s commented on tests/testthat/test_use_cases.R:
```diff
@@ -9,46 +9,48 @@ data <- prostate
+    , c("stabpath", "matrix"))
+  expect_named(
+    get_stability_paths(model, data, reps = 10, method = "bootstrap")[5, ]
+    , c("lcavol", "lweight", "age", "lbph", "svi", "lcp", "gleason", "pgg45"))
```
plz fix the formatting

fabian-s commented on tests/testthat/test_use_cases.R:
```diff
@@ -9,46 +9,48 @@ data <- prostate
+test_that("get_stability_paths with prostate data - subsample, row values", {
+  expect_equal(
+    unname(s)
+    , c(1.0, 0.8, 0.3, 0.7, 0.9, 0.1, 0.0, 0.2))
```
plz fix the formatting

fabian-s commented on tests/testthat/test_use_cases.R:
```diff
@@ -9,46 +9,48 @@ data <- prostate
+    , c("stabpath", "matrix"))
+  expect_named(
+    get_stability_paths(model, data, reps = 10)[5, ]
+    , c("lcavol", "lweight", "age", "lbph", "svi", "lcp", "gleason", "pgg45"))
```
plz fix the formatting

fabian-s commented on tests/testthat/test_use_cases.R:
```diff
@@ -9,46 +9,48 @@ data <- prostate
 max_formula <- lpsa ~ (. - train)
 model <- regsubsets(max_formula, data=data, nbest = 1, nvmax = 8,
```
plz fix: `max_formula` is not found when this is run via `check()`-- do we really need to define a separate variable for the formula? this might work if we supply the formula directly, the indirection here might be the problem when this gets evaluated in the weird testthat-sandbox environment....

fabian-s commented on tests/testthat/test_use_cases.R:
```diff
@@ -9,46 +9,48 @@ data <- prostate
+test_that("get_stability_paths with prostate data - subsample", {
+  expect_is(
+    get_stability_paths(model, data, reps = 10)
+    , c("stabpath", "matrix"))
```
plz fix the formatting

fabian-s commented on tests/testthat/test_use_cases.R:
```diff
@@ -9,46 +9,48 @@ data <- prostate
+test_that("get_stability_paths with prostate data - bootstrap", {
+  expect_is(
+    get_stability_paths(model, data, reps = 10, method = "bootstrap")
+    , c("stabpath", "matrix"))
```
plz fix the formatting

fabian-s commented on tests/testthat/test_use_cases.R:
```diff
@@ -9,46 +9,48 @@ data <- prostate
+test_that("get_stability_paths with prostate data - bootstrap, row values", {
+  expect_equal(
+    unname(b)
+    , c(1.0, 0.7, 0.4, 0.4, 0.9, 0.2, 0.2, 0.2))
```
plz fix the formatting

Commit:
Fix of "issue #4: Test infrastructure and first tests generated"

Following tools are running without issues:
devtools::test()
covr::package_coverage()
devtools::check()
devtools::build()

Nevertheless R CMD check / check() provides following note:
checking R code for possible problems ... NOTE
plot.stabpath: no visible global function definition for 'par'
plot.stabpath: no visible binding for global variable 'par'
plot.stabpath: no visible global function definition for 'modifyList'
plot.stabpath: no visible global function definition for 'hcl'
plot.stabpath: no visible binding for global variable 'matplot'
plot.stabpath: no visible binding for global variable 'mtext'
Undefined global functions or variables:
  hcl matplot modifyList mtext par
Consider adding
  importFrom("grDevices", "hcl")
  importFrom("graphics", "matplot", "mtext", "par")
  importFrom("utils", "modifyList")
to your NAMESPACE file.

Since from my pov note item isindependent from fixing issue #4 I have not fixed the note item.

fabian-s commented:
Thanks!
